cmake/Modules/FindLibsecret.cmake

```diff
@@ -0,0 +1,111 @@
+#[=======================================================================[.rst
+FindLibsecret
+--------
+
+FindModule for libsecret and the associated library
+
+Imported Targets
+^^^^^^^^^^^^^^^^
+
+.. versionadded:: 2.0
+
+This module defines the :prop_tgt:`IMPORTED` target ``Libsecret::Libsecret``.
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+This module sets the following variables:
+
+``Libsecret_FOUND``
+  True, if the library was found.
+``Libsecret_VERSION``
+  Detected version of found Libsecret library.
+
+Cache variables
+^^^^^^^^^^^^^^^
+
+The following cache variables may also be set:
+
+``Libsecret_INCLUDE_DIR``
+  Directory containing ``secret.h``.
+
+#]=======================================================================]
+
+include(FindPackageHandleStandardArgs)
+
+find_package(PkgConfig QUIET)
+if(PKG_CONFIG_FOUND)
+  pkg_search_module(PC_Libsecret QUIET libsecret-1)
+endif()
+
+# Libsecret_set_soname: Set SONAME on imported library target
+macro(Libsecret_set_soname)
+  if(CMAKE_HOST_SYSTEM_NAME MATCHES "Linux")
+    execute_process(
+      COMMAND sh -c "objdump -p '${Libsecret_LIBRARY}' | grep SONAME"
```

Comment:
We haven't updated that in the other finders yet (but we should): Use `${CMAKE_OBJDUMP}` here for cases where a toolchain file is provided, which might set an architecture-specific `objdump` binary.